### ISSUE TITLE
Edit file with key 'e' opens editor on wrong line number

### ISSUE THREAD

**Issue by andenkondor** [jesseduffield/lazygit#3337]
**Describe the bug**
In some circumstances pressing 'e' on a file in stage view to open it, opens the file with the cursor on the wrong line number.

**To Reproduce**
1. Create a file with following content and commit it:

```
1
2
3




A
B
C
D
E
F



%
$
%
^
```

2. Now do changes in the file so it looks like this:
```
1
2
new line
3




A
B
C
D
E
F



%
$
%
^
(

```

3. Now open lazygit 
4. Go to 'Files'
5. Hit Enter to go to staging editor
6. Move the cursor to '%' and press 'e'
7. In the opened editor (for me it's nvim) not the '%' line is selected but the '$' line.

**Expected behavior**
When I press 'e' on a line XY, the opened editor will have the cursor pointed to line XY.

**Version info:**
commit=, build date=, build source=homebrew, version=0.40.2, os=darwin, arch=arm64, git version=2.42.0
git version 2.42.0

**Additional context**
The problem is that lazyvim will select the line number from where the change happened in the old file.
But if there are also hunks with additions/deletions before that line, these have to be involved.
The 'E' option for editing the hunk is working perfectly fine, here the cursor is always places correctly.

**Comment by stefanhaller** [comment]
This was already fixed last August (#2919). Sorry that it takes so long to make a new release, we'll hopefully have one relatively soon.

**Comment by andenkondor** [comment]
@stefanhaller Didn't notice that one. Thank you! Really looking forward to the new release then.
